File: game-server/src/main/java/com/kingston/mmorpg/game/ai/monster/DamageStatistics.java
package com.kingston.mmorpg.game.ai.monster;

import java.util.HashMap;
import java.util.Map;

/**
 * 怪物的伤害统计
 * 
 * @author kingston
 *
 */
public class DamageStatistics {
	
	/**
	 * 统计列表 攻击方id -> 伤害值
	 */
	public Map<Long, Long> damages = new HashMap<>();
	
	public void addDamage(long attacker, long hurt) {
		this.addDamage(attacker, hurt);
	}
	
	public long getDamage(long attacker) {
		return damages.getOrDefault(attacker, 0L);
	}
	
	public void clearDamage() {
		this.damages.clear();
	}

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/ai/monster/DamageStatistics.kt
package org.forfun.mmorpg.game.ai.monster

/**
 * 怪物的伤害统计
 */
class DamageStatistics {

    /**
     * 统计列表 攻击方id -> 伤害值
     */
    val damages = HashMap<Long, Long>()

    fun addDamage(attacker: Long, hurt: Long) {
        this.damages[attacker] = this.damages.getOrDefault(attacker, 0L) + hurt
    }

    fun getDamage(attacker: Long): Long {
        return damages.getOrDefault(attacker, 0L)
    }

    fun clearDamage() {
        this.damages.clear()
    }
}
